tasks/firewall.yml

```diff
@@ -25,11 +25,19 @@
     msg: "Both FirewallD and UFW are detected, firewall situation is unknown"
   when: openvpn_firewall == 'auto' and firewalld.rc == 0 and ufw.rc == 0
 
-- name: Fail on no firewall detected
-  fail:
-    msg: "No firewall detected, install one before proceeding (firewalld||ufw||iptables)"
+- name: Install firewalld if no firewall detected
+  package:
+    name: firewalld
+    state: present
   when: firewalld.rc != 0 and ufw.rc != 0 and iptables.rc != 0
 
+- name: Check for firewalld
+  shell: command -v firewall-cmd
```

Comment:
just an fyi - `which` is used earlier because it's an external binary, it doesn't require shell